examples/cluster/plot_hierarchical_clustering_dendrogram.py

```diff
@@ -0,0 +1,45 @@
+# Authors: Mathew Kallada
+# License: BSD 3 clause
+"""
+=========================================
+Plot Hierarachical Clustering Dendrogram 
+=========================================
+
+This example plots the corresponding dendrogram of a hierarchical clustering
+using AgglomerativeClustering and the dendrogram method available in scipy.
+"""
+
+import numpy as np
+
+from matplotlib import pyplot as plt
+from scipy.cluster.hierarchy import dendrogram
+from sklearn.datasets import load_iris
+from sklearn.cluster import AgglomerativeClustering
+
+def plot_dendrogram(model, **kwargs):
+
+    # Children of hierarchical clustering
+    children = model.children_
+
+    # Distances between each pair of children
+    # Since we don't have this information, we can use a uniform one for plotting
+    distance = np.arange(children.shape[0])
+
+    # The number of observations contained in each cluster level
+    no_of_observations = np.arange(2, children.shape[0]+2)
```

Comment:
i think this is not what is expected by scipy's `dendrogram` method:

http://docs.scipy.org/doc/scipy/reference/generated/scipy.cluster.hierarchy.linkage.html#scipy.cluster.hierarchy.linkage

> The fourth value Z[i, 3] represents the number of original observations in the newly formed cluster.

Just a counter example: if in the first two iterations the algorithm decides to merge the original observations with indices (0, 1) and (2, 3) both of the newly formed clusters then would have a size of 2. That's what `dendrogram` seems to expect (which `arange` doesn't offer).

Currently i don't see an easy way of accessing that information in sklearn.